aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -83,21 +84,21 @@ public boolean hasPendingMutation(@NonNull String modelId) {
         String modelId = incomingMutation.getMutatedItem().getId();
         @SuppressWarnings("unchecked")
         PendingMutation<T> existingMutation = (PendingMutation<T>) nextMutationForModelId(modelId);
-        if (existingMutation == null || inFlightMutations.contains(existingMutation.getMutationId())) {
-            return save(incomingMutation).andThen(notifyContentAvailable());
-        }
-        switch (incomingMutation.getMutationType()) {
-            case CREATE:
-                return handleCreation();
-            case UPDATE:
-                return handleUpdate(existingMutation, incomingMutation);
-            case DELETE:
-                return handleDeletion(existingMutation, incomingMutation);
-            default:
-                return unknownMutationType(incomingMutation.getMutationType());
+        if (existingMutation == null) {
+            return save(incomingMutation)
+                .andThen(notifyContentAvailable());
+        } else {
+            return resolveConflict(existingMutation, incomingMutation);
```

Comment:
It looks like before, it would always handle the mutation type but now you're only handling it if the existingMutation is null so just wanted to check that's intended behavior.

Reply:
I encapsulated that logic in the `IncomingMutationConflictHandler`. That class takes in both the existing and incoming mutation and applies logic similar to how iOS and JS are doing right now...at east that was the goal :)

The entry point for that is [right here](https://github.com/aws-amplify/amplify-android/pull/498/files#diff-7fc8fd67eed2c76db4c431af072805ccR293)